### ISSUE TITLE
SPO site not deployed with package

### ISSUE THREAD

**Issue by LisaJo48** [OfficeDev/microsoft-teams-apps-champion-management#117]
Reviewed all the notes from the open ticket "Spinner...". Deinstalled, removed TEAMS app, deleted package from app store, redownloaded package, reuploaded package MULTIPLE times. All the install processes appear to function correctly; no errors. Both APIs are there and approved. Get this when I click on the icon in TEAMS (and BTW, I don't get the "Add" button; it just opens the app) and it starts spinning and I get this error message. Waited overnight: no change. Refreshed multiple times: no change. Left my computer on and running with the spinner going for several hours: no change.
![CMP Stuck](https://user-images.githubusercontent.com/46016140/185769250-0526c0d1-a1e8-48ec-885a-8c2f14a329af.png)

Further investigation revealed there's no SPO site created. The screenshot below is from my last attempt minutes ago. The new site for CMP should be at the top (sorted new to old per create date). It is not there.

![image](https://user-images.githubusercontent.com/46016140/185769280-dc6a92a1-c691-4372-8844-739462bfe8c1.png)

I have no idea what to do next. Please help.

**Comment by v-saikirang** [comment]
Hi @LisaJo48,

Thanks for reaching out to us. 
Can you please check if the site is under 'Deleted sites'in SP Admin center. If yes please delete it from there as well and try opening the app after 15 minutes.

Can you also please confirm if you are using admin account to open the app in teams.

Thanks.

**Comment by LisaJo48** [comment]
I deleted the site, permanently, via the SP Admin portal using an account with the SP Admin role. At the moment, all components have been removed: site, TEAM app, package, and APIs.

Additional info: I tried the whole process on a different computer (same account) and got the same results. I also tried it using another Admin account: same results.

**Comment by v-saikirang** [comment]
Hi @LisaJo48,

In the other ticket #116 that you opened you shared a screenshot of the app. So I assume you were able to install the app once. May I know if that is a different tenant.
Thanks.

**Comment by LisaJo48** [comment]
Same tenant. I had done some testing in my first install and, therefore, wanted to 'start fresh' and removed the package et al and had this outcome the next time.

**Comment by LisaJo48** [comment]
New experience. I attempted to install it this evening (Microsoft reported a 'problem' with ODB and TEAMS today that was resolved mid-day) and, now, I am unable to approve both APIs. User.ReadWrite approved as expected; Sites.Manage.All did not. I received this error:
![image](https://user-images.githubusercontent.com/46016140/186046410-22a01d41-3776-421a-8a76-e47900c5dd6f.png)

This is the first time I have had this result. I refreshed and completely closed the browser and tried again: received the same error. Therefore, I have removed the package and the 2 APIs.

**Comment by v-saikirang** [comment]
Hi @LisaJo48,

The Sites.Manage.All is required to install the app. So please clear out the APIs and re deploy the package and approve those 2 permissions. 
And please double check "Deleted Sites" section in admin center to make sure "championmanagementplatform" is still not hanging in there. 
The error in "API access" page might be a glitch. We do not have control on that. But we could try the process again.
We will not close this ticket until you confirm. After trying the above steps if it still doesn't work we can try to troubleshoot over a Teams call. Please let me know.

Thanks.

**Comment by LisaJo48** [comment]
As before, there is not deleted SPO site to permanently remove. I also tried remove-SPOsite and Remove-SPODeletedSite in PowerShell. I was able to approve the API that wouldn't approve yesterday (screenshot above) and then deleted it along with the other. Then deleted the package again and deleted the TEAM App. Then tried to install again: downloaded package, uploaded package, approve APIs and, again, was only able to approve one. The Sites.Manage.All again will not approve (same error). Could go no further with the process.

Question: which step provisions the website? Is it related to this API that won't be approves or something else? Is it possible for a GA to restrict which admins (i.e. SharePoint Admin) from being able to approve [some] APIs? I don't know of any changes made to the tenant between when I first installed and when I've attempted to reinstall.

I believe a call would be helpful. Maybe I'm missing something. The earliest opportunity to schedule a meeting will be Monday, August 29 between 3 and 5 pm EASTERN.

Thank you.

**Comment by v-saikirang** [comment]
Hi @LisaJo48,

Yes, Sites.Manage.All is the permission required for the app to create the sharepoint site. And the site is created when you open the app after the permission is approved. On load of the app If the site is not present it creates the site collection. You were once able to approve and some times not. So I suggest you re try to see if you can approve this permissions. Delete the Sites.Manage.All permission, delete the package and reupload the package and see if you can approve. You dont have to remove the other permission. If you are still not able to approve then please check with your GlobalAdmin/Sharepoint admin. The app itself has nothing to do with the permission approval. So it should be a glitch related to "API Access" in SP admin center.
Please let me know how this turns up and then we can have a call.
Thanks.

**Comment by v-saikirang** [comment]
Hi @LisaJo48 ,

Thanks for your patience on this.

Now that the Api permissions are approved there is no issue with the admin center.

Can you please open the app in browser and share the console logs. I will verify the logs and we can set up a call if needed.
Thanks.

**Comment by LisaJo48** [comment]
WELL! I ran Get-SPOsite command first and found the site was there. So, I opened the app in the browser and, low and behold, it loaded! Then I rechecked the SPO Admin portal and now it's listed there, too, with a create date of August 31st! I promise you, it was NOT there when I last checked that same day.

For what it's worth, the ONLY thing I did differently this time (from the multiple attempts prior over the last 2 weeks) is that I used the Get command.  However unlikely is sounds, running the Get command seems to have wriggled something loose, so the speak, so that the command either finally finished provisioning the site and recorded the create date as the day of my last attempt that didn't work (noted above), or the site _had_ fully provisioned itself on August 31st but wasn't displayed in the admin portal until today when I tried the Get command. 

In the end, weird and maybe inexplicable, but thanks for your continued attention. Without your willingness to stick with me on this, I don't think I would have been successful on my own.
